### PR TITLE
[Gutenberg] - React Native 0.69.4 Upgrade

### PR DESCRIPTION
**Pull request by geriux**
This PR adds changes to support the React Native `0.69.4` upgrade.

It removes the reference to `hermes-mirror` since it's no longer used.

It also adds several implementations in `libs/editor/build.gradle` since they're not included with the React Native and Hermes binaries.

### Feature branch

- [Gutenberg Mobile](https://github.com/wordpress-mobile/gutenberg-mobile/pull/5193)
- [Gutenberg](https://github.com/WordPress/gutenberg/pull/43485)

### Android integration

- [Gutenberg Mobile](https://github.com/wordpress-mobile/gutenberg-mobile/pull/5170)
- [Gutenberg](https://github.com/WordPress/gutenberg/pull/43486)
- [WordPress Android](https://github.com/wordpress-mobile/WordPress-Android/pull/17066)

### iOS integration

- [Gutenberg Mobile](https://github.com/wordpress-mobile/gutenberg-mobile/pull/5192)
- [Gutenberg](https://github.com/WordPress/gutenberg/pull/44889)
- [WordPress iOS](https://github.com/wordpress-mobile/WordPress-iOS/pull/19326)

## To test:
CI checks should pass, as well as checking the editor opens correctly with the builds from this PR.

## Regression Notes
1. Potential unintended areas of impact
Editor

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing, CI tests.

3. What automated tests I added (or what prevented me from doing so)
None as there is no added functionality

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/build.gradle
Comment (geriux) [inline WordPress/build.gradle]: Todo: How to exclude this in release builds? If we exclude them like this, hermes is no longer being included in the build.

**Comment by fluiddot** [comment]
@geriux @derekblank I noticed that some PR checks failed, have you managed to check if the failure is uncovering an actual issue?

Regarding the `instrumented-tests` check, I saw that one of the tests is crashing ([reference](https://buildkite.com/automattic/wordpress-android/builds/7150#0183f009-ca4f-4b9e-a37e-a4302d3ed180/322-726)), it would be great to check it in case there's something going on. Thanks!

**Comment by geriux** [comment]
@fluiddot Android CI issues are now solved after https://github.com/wordpress-mobile/WordPress-Android/pull/17066/commits/5dad8d585c0f0b523f34c951dad614970b3be383

File: build.gradle
Comment (fluiddot) [inline build.gradle]: I think there's a typo here as the Git tag for [the alpha version is `v1.86.0-alpha1`](https://github.com/wordpress-mobile/gutenberg-mobile/releases/tag/v1.86.0-alpha1), there's an extra `.` at the beginning of the string.

**Review by fluiddot — CHANGES_REQUESTED** [review]
@derekblank I reviewed the PR and noticed an issue in the latest code changes that might be the culprit of the CI checks nos passing, although it's hard to tell based on [the error output](https://buildkite.com/automattic/wordpress-android/builds/8101#0184e202-70cf-4208-8e4e-5e07cc8cf749/176-179). Before proceeding, let's try to solve them and check if the CI checks pass.

File: build.gradle
Comment (derekblank) [reply build.gradle]: Thanks for pointing this out! There was also a dash between `alpha` and `1`. 😅 I've made some updates, and the CI is now passing. 👍

File: WordPress/build.gradle
Comment (fluiddot) [inline WordPress/build.gradle:37]: > It removes the reference to hermes-mirror since it's no longe

I understand that now on we won't need to publish the hermes artifacts as we were doing [here](https://github.com/wordpress-mobile/react-native-mirror-publisher/blob/50d840389524203b22ed2b51e2b28c67f4b9b942/build.gradle.kts#L23-L51) in the `react-native-mirror-publisher` repository, is this accurate? If so, we might update that repository as a follow-up task once this gets merged.

**Comment by geriux** [comment]
CI checks are now green and the installable build opens the editor correctly.

File: libs/editor/build.gradle
Comment (fluiddot) [inline libs/editor/build.gradle:83]: > It also adds several implementations in libs/editor/build.gradle since they're not included with the React Native and Hermes binaries.

Are these the libraries not included in the React Native and Hermes binaries? I'm wondering if we'd need to define them here or if they could be part of the libraries required by Gutenberg Mobile, could we have them as [dependencies in the `react-native-bridge` build gradle file](https://github.com/WordPress/gutenberg/blob/3b04ce46f18d61aa25f0483ba49116e701835fdd/packages/react-native-bridge/android/react-native-bridge/build.gradle#L59)?

I noticed that we are also setting the React Native version, is this required? Would be a way to let the Gutenberg Mobile library to define this?

File: libs/editor/build.gradle
Comment (fluiddot) [inline libs/editor/build.gradle:93]: I think we haven't updated the Aztec version, so I'm wondering why we need to add this dependency if we didn't have it before 🤔, what changed with the RN upgrade to producing this?

File: WordPress/build.gradle
Comment (geriux) [reply WordPress/build.gradle:37]: >I understand that now on we won't need to publish the hermes artifacts as we were doing [here](https://github.com/wordpress-mobile/react-native-mirror-publisher/blob/50d840389524203b22ed2b51e2b28c67f4b9b942/build.gradle.kts#L23-L51) in the react-native-mirror-publisher repository, is this accurate?

No, it is not. We are publishing the Hermes artifacts but under the `react-native` [repository](https://github.com/wordpress-mobile/react-native-mirror-publisher/commit/006b8c05886f0089c2071b388beff29b2f6483c3) and not the `hermes-mirror` one we used before.

Comment (fluiddot) [reply WordPress/build.gradle:37]: Ah great, thanks for the clarification 🙇.

**Review by fluiddot — APPROVED** [review]
I tested the installable build and the editor worked as expected. LGTM 🎊 !

Regarding [the comments I posted in my last review](https://github.com/wordpress-mobile/WordPress-Android/pull/17066/#pullrequestreview-1208027181), I think we should address them. However, I agree that shouldn't block the merge, hence we could tackle them in a follow-up PR if needed. Thanks!

File: libs/editor/build.gradle
Comment (geriux) [reply libs/editor/build.gradle:83]: Since we encountered issues with using the `debug` artifacts for both Hermes and React Native instead of the `release` ones, when excluding them and including them again some libraries needed to be referenced again.

I'm working on a draft PR as a follow-up to improve these, although to be able to force the release artifacts for all builds like (Jalapeno Debug) which are the builds PRs generate, we will have to keep a few implementations. 😅

File: libs/editor/build.gradle
Comment (geriux) [reply libs/editor/build.gradle:93]: This [comment](https://github.com/wordpress-mobile/WordPress-Android/pull/17066/files#r1042123549) is also related for this question.

Comment (ParaskP7) [reply libs/editor/build.gradle:93]: 👋 @geriux @fluiddot !

I also stumbled upon this just now, that is, while working on the `org.jsoup:jsoup` dependency update (see [here](https://github.com/wordpress-mobile/WordPress-Android/issues/17569)). I just removed this dependency from the `libs:editor` module (it being unused and all) and the WPAndroid seems to be building/working just fine.

> I'm working on a draft PR as a follow-up to improve these, although to be able to force the release artifacts for all builds like (Jalapeno Debug) which are the builds PRs generate, we will have to keep a few implementations. 😅

Should I refrain from removing dependency and ignore this for now, that is, at least until this work above is done? Or maybe it is already done but I can't connect the dots just yet, not sure? 🤔

Thanking you in advance for your advise here! 🙏

Comment (geriux) [reply libs/editor/build.gradle:93]: > and the WPAndroid seems to be building/working just fine.

Does the editor open correctly after this change?

Comment (ParaskP7) [reply libs/editor/build.gradle:93]: 👋 @geriux and thanks for the reply!

> Does the editor open correctly after this change?

Yes, I am able to open the editor correctly after this change (both the `Classic` and `Block` editor). Unless of course, I am missing something obvious here... 🤔

Comment (ParaskP7) [reply libs/editor/build.gradle:93]: FYI: I just pushed this [removal commit](https://github.com/wordpress-mobile/WordPress-Android/pull/18102/commits/e98d7e8f729084eabe8911943bb28796f3ac5c89), feel free to test [this PR](https://github.com/wordpress-mobile/WordPress-Android/pull/18102) yourself and let me know if everything works as you would have expected.

Comment (fluiddot) [reply libs/editor/build.gradle:93]: Looks like the dependency `org.jsoup:jsoup` is used by Aztec-Android ([reference](https://github.com/wordpress-mobile/AztecEditor-Android/blob/23983f83a710ba5471c79416e6ce96febc63d6b9/aztec/build.gradle#L52)).

In fact, we set the version of that dependency [here in Gutenberg](https://github.com/WordPress/gutenberg/blob/b3a66dbc044d8c273744520161ea6436e07c1aaf/packages/react-native-aztec/android/build.gradle#L9). I'm not sure whether it can be removed or not, in any case, this could be tested by checking if the editor can be built and works as expected.

**Pull request merged**
